pkg/segment/search/searchaggs.go

```diff
@@ -553,66 +553,61 @@ func PerformMeasureAggsOnRecs(nodeResult *structs.NodeResult, recs map[string]ma
 		firstRecInden = recInden
 		break
 	}
+	bb := bbp.Get()
+	defer bbp.Put(bb)
+
+	aggCols, aggColUsage, valuesUsage, listUsage := getSegStatsMeasureCols(nodeResult.MeasureOperations)
 
 	for recInden, record := range recs {
 		sstMap := make(map[string]*structs.SegStats, 0)
-
-		for _, mOp := range nodeResult.MeasureOperations {
-			dtypeVal, err := utils.CreateDtypeEnclosure(record[mOp.MeasureCol], qid)
+		for column := range aggCols {
+			if column == "*" {
+				stats.AddSegStatsCount(sstMap, column, 1)
+				continue
+			}
```

Comment:
`column == "*"` only makes sense for the `Count` aggregation, so I think it would be good to double check that here and log an error if we have a different aggregation for `column == "*"`. Or is there a reason you're not doing that?

Reply:
The check is done in updateSegmentStats, which PerformAggs calls to update the segment stats.